### DISCUSSION TITLE
Streams use MTU?

### DISCUSSION THREAD

**Discussion by okbaswc** [microsoft/msquic#2276] [Q&A] [answered]
Do I need to break up my buffers/sends over a stream into MTU sizes for most efficient transport? Or does MSQUIC do this internally? E.G. I send a QUIC_BUFFER with 20kb of data, does it send in MTU sized packets automatically?

Also do datagrams have lower latency than streams?

**Comment by nibanks** [comment, marked as answer]
> Do I need to break up my buffers/sends over a stream into MTU sizes

No, that is all done automatically by MsQuic.

> Also do datagrams have lower latency than streams?

Only very slightly most of the time, but we are currently doing a lot of work to try to make streams the absolute lowest latency. The only place datagrams are really different is in the face of packet loss; they won't be retransmitted. You have the choice to do that or not when they're lost.
